### PR TITLE
[IMP] Add already the possibility to fill in the products you want in an...

### PR DESCRIPTION
**Pull request by jco-odoo**
... inventory

In order to do that, we change the theoretical quantity into a functional stored field.
Therefore the on_change changes, but the old still work.
The UoM of the inventory line is also taken into account

@odony Changes should be as such that it won't give a problem for the existing 8.0 (The tests I made, were fine)

### CODE REVIEW

File: addons/stock/stock.py [hunk 1]
```diff
@@ -2735,11 +2738,7 @@ def prepare_inventory(self, cr, uid, ids, context=None):
                 # On import calculate theoretical quantity
                 quant_obj = self.pool.get("stock.quant")
                 for line in inventory.line_ids:
```
Comment (odony) [inline addons/stock/stock.py]: this whole loop is redundant now that `theoretical_qty` is a function

File: addons/stock/stock.py [hunk 2]
```diff
@@ -2571,7 +2571,7 @@ def _get_available_filters(self, cr, uid, context=None):
            :rtype: list of tuple
         """
         #default available choices
-        res_filter = [('none', _('All products')), ('product', _('One product only'))]
+        res_filter = [('none', _('All products')), ('partial', _('Partial')), ('product', _('One product only'))]
```
Comment (odony) [inline addons/stock/stock.py]: how about `('partial', _('Manual selection of products'))` ?  Keeping in mind that the label is "Inventory of"

File: addons/stock/stock.py [hunk 3]
```diff
@@ -2614,7 +2614,10 @@ def _get_total_qty(self, cr, uid, ids, field_name, args, context=None):
         'partner_id': fields.many2one('res.partner', 'Inventoried Owner', readonly=True, states={'draft': [('readonly', False)]}, help="Specify Owner to focus your inventory on a particular Owner."),
         'lot_id': fields.many2one('stock.production.lot', 'Inventoried Lot/Serial Number', readonly=True, states={'draft': [('readonly', False)]}, help="Specify Lot/Serial Number to focus your inventory on a particular Lot/Serial Number.", copy=False),
         'move_ids_exist': fields.function(_get_move_ids_exist, type='boolean', string=' Stock Move Exists?', help='technical field for attrs in view'),
-        'filter': fields.selection(_get_available_filters, 'Selection Filter', required=True),
+        'filter': fields.selection(_get_available_filters, 'Selection Filter', required=True,
```
Comment (odony) [inline addons/stock/stock.py]: - update help if label changed to 'Manual selection of products'
- fix `string` and set it to "Inventory of" for consistency with the view

File: addons/stock/stock.py [hunk 4]
```diff
@@ -2810,7 +2822,9 @@ def _get_prodlot_change(self, cr, uid, ids, context=None):
         'company_id': fields.related('inventory_id', 'company_id', type='many2one', relation='res.company', string='Company', store=True, select=True, readonly=True),
         'prod_lot_id': fields.many2one('stock.production.lot', 'Serial Number', domain="[('product_id','=',product_id)]"),
         'state': fields.related('inventory_id', 'state', type='char', string='Status', readonly=True),
-        'theoretical_qty': fields.float('Theoretical Quantity', digits_compute=dp.get_precision('Product Unit of Measure'), readonly=True),
+        'theoretical_qty': fields.function(_get_theoretical_qty, type='float', digits_compute=dp.get_precision('Product Unit of Measure'),
+                                           store={'stock.inventory.line': (lambda self, cr, uid, ids, c={}: ids, ['product_id', 'product_uom_id', 'location_id', 'package_id', 'prod_lot_id', 'company_id', 'partner_id'], 20),},
```
Comment (odony) [inline addons/stock/stock.py]: `store=True` would be simpler and work just as well?

File: addons/stock/stock.py [hunk 5]
```diff
@@ -2861,23 +2906,6 @@ def _resolve_inventory_line(self, cr, uid, inventory_line, context=None):
         return stock_move_obj.create(cr, uid, vals, context=context)
 
     def restrict_change(self, cr, uid, ids, theoretical_qty, context=None):
```
Comment (odony) [inline addons/stock/stock.py]: add a comment on both `restrict_change` and `on_change_product_id` to indicate they should be deleted in saas-6/master

File: addons/stock/stock.py [hunk 6]
```diff
@@ -2827,9 +2841,40 @@ def _get_prodlot_change(self, cr, uid, ids, context=None):
...
     }
 
+    def _get_quants(self, cr, uid, line, context=None):
+        quant_obj = self.pool.get("stock.quant")
+        dom = [('company_id', '=', line.company_id.id), ('location_id', 'child_of', line.location_id.id), ('lot_id', '=', line.prod_lot_id.id),
+                        ('product_id','=', line.product_id.id), ('owner_id', '=', line.partner_id.id), ('package_id', '=', line.package_id.id)]
+        quants = quant_obj.search(cr, uid, dom, context=context)
+        return quants
+
+    def onchange_createline(self, cr, uid, ids, location_id=False, product_id=False, uom_id=False, package_id=False, prod_lot_id=False, partner_id=False, company_id=False, context=None):
```
Comment (odony) [inline addons/stock/stock.py]: in terms of code cleaning:
- self.pool.get('x') -> self.pool['x']
- dic.update({'foo': 'bar'}) -> dic['foo'] = 'bar'

**Comment by jco-odoo** [comment]
Merged

**Pull request closed without merging**